src/plugins/pcr_primer_design_for_dna_assembly/src/PCRPrimerDesignForDNAAssemblyOPWidget.cpp

```diff
@@ -282,4 +286,66 @@ void PCRPrimerDesignForDNAAssemblyOPWidget::sl_loadOtherSequenceInPcr() {
 
 }
 
+void PCRPrimerDesignForDNAAssemblyOPWidget::sl_onFindTaskFinished() {
+    CHECK(sender() == pcrTask, );
+    SAFE_POINT(nullptr != pcrTask, L10N::nullPointerError("InSilicoPcrTask"), );
+    if (pcrTask->isCanceled() || pcrTask->hasError()) {
+        disconnect(pcrTask, SIGNAL(si_stateChanged()));
+        pcrTask = nullptr;
+        setEnabled(true);
+        return;
+    }
+    CHECK(pcrTask->isFinished(), );
+    //do nothing in case of empty result
+    foreach (const U2Region &region, pcrTask->getResults()) {
+        if (region != U2Region()) {
+            showResults();
+            createResultAnnotations();
+            pcrTask = nullptr;
+            setEnabled(true);
+            return;
+        }
+    }
+}
+
+void PCRPrimerDesignForDNAAssemblyOPWidget::showResults() {
+    productsTable->show();
+    productsTable->setCurrentProducts(pcrTask->getResults(), annDnaView);
+}
+
+void PCRPrimerDesignForDNAAssemblyOPWidget::createResultAnnotations() {
+    QString pcrTableObjectName = tr("PCR Primer Design for DNA assembly");
+    QStringList usedNames;
+    for (AnnotationTableObject *tableObject : annDnaView->getActiveSequenceContext()->getAnnotationObjects()) {
+        usedNames.append(tableObject->getGObjectName());
+    }
+    int counter = 1;
+    QString rolledName = pcrTableObjectName;
+    while (usedNames.contains(pcrTableObjectName)) {
+        rolledName = pcrTableObjectName + QString(" %1").arg(counter);
+        counter++;
+    }
+
+    U2OpStatusImpl os;
+    const U2DbiRef localDbiRef = AppContext::getDbiRegistry()->getSessionTmpDbiRef(os);
+    SAFE_POINT_OP(os, );
+    AnnotationTableObject *newDocAto = new AnnotationTableObject(rolledName, localDbiRef);
+    QList<SharedAnnotationData> annotations;
+    int index = 0;
+    foreach (const U2Region &region, pcrTask->getResults()) {
+        if (region != U2Region()) {
+            SharedAnnotationData data(new AnnotationData());
+            data->setStrand(index % 2 == 0 ? U2Strand(U2Strand::Direct) : U2Strand(U2Strand::Complementary));
+            data->name = PCRPrimerDesignForDNAAssemblyTask::FRAGMENT_INDEX_TO_NAME.at(index);
+            data->location->regions.append(region);
+            annotations.append(data);
+        }
+        index++;
+    }
+    newDocAto->addAnnotations(annotations, tr("PCR Primer Design for DNA assembly"));
```

Comment:
You already have a variable `pcrTableObjectName` with the value `tr("PCR Primer Design for DNA assembly")`